src/postamble.js

```diff
@@ -397,7 +397,7 @@ function exit(status, implicit) {
   // don't need to do anything here and can just leave. if the status is
   // non-zero, though, then we need to report it.
   // (we may have warned about this earlier, if a situation justifies doing so)
-  if (implicit && noExitRuntime && status === 0) {
+  if (implicit && status === 0) {
```

Comment:
this has the consequence that if we are exiting the runtime with 0, we just return here, and don't reach `exitRuntime()` which is called lower down. I think that's wrong, and I'd expect a test to fail due to it, but let's see if I'm missing something when CI finishes...

Reply:
Ah, I missed the call to `exitRuntime`. You're right that this will probably fail. However, without this change, when `noExitRuntime` is falsey (so the exit runtime is enabled), `exit` never returns so the postRun hooks are never called. I suppose a more correct solution would be to move the postRun hook execution into this function? I suspect doing that might break something else, though. Do you have any ideas?

Reply:
What postRun hook do we need?

Reply:
The `test_embind_no_rtti*` tests run some code in a postRun hook. Since lsan enables the exit runtime, the hook is never called and the tests fail. I guess arguably the tests should not be calling exported functions in a postRun hook because the C++ runtime has been torn down, but it still seems counterintuitive that postRun hooks are not called when the exit runtime is enabled.

Reply:
Hmm, I don't see docs for postRun, so I'm actually not sure what it formally is supposed to do. I agree that is a little confusing, but I'm not sure there isn't a good reason for it.

It might be best to refactor the test to avoid postRun. (Perhaps we should even look into removing it?)

Reply:
Sure, I'll do that refactoring and open a separate issue about postRun.